cmd/lk/app.go

```diff
@@ -0,0 +1,379 @@
+		if err != nil {
+			return err
+		}
+		templateSelect := huh.NewSelect[string]().
+			Title("Select Template").
+			Value(&templateURL).
+			WithTheme(theme)
+		var options []huh.Option[string]
+		for _, t := range templates {
+			options = append(options, huh.NewOption(t.Name, t.URL))
+		}
+		templateSelect.(*huh.Select[string]).Options(options...)
+		preinstallPrompts = append(preinstallPrompts, templateSelect)
+		// if templateName is specified, fetch template index and find it
+	} else if templateName != "" {
+		templates, err := bootstrap.FetchTemplates(ctx)
+		if err != nil {
+			return err
+		}
+		for _, t := range templates {
+			if t.Name == templateName {
+				template = &t
+				templateURL = t.URL
+				break
+			}
+		}
+		if template == nil {
+			return errors.New("template not found: " + templateName)
+		}
+	}
+
+	if len(preinstallPrompts) > 0 {
+		group := huh.NewGroup(preinstallPrompts...)
+		if err := huh.NewForm(group).
+			WithTheme(theme).
+			RunWithContext(ctx); err != nil {
+			return err
+		}
+	}
+
+	fmt.Println("Cloning template...")
+	if err := cloneTemplate(ctx, cmd, templateURL, appName); err != nil {
+		return err
+	}
+
+	fmt.Println("Instantiating environment...")
+	if err := instantiateEnv(ctx, cmd, appName); err != nil {
+		return err
+	}
+
+	fmt.Println("Installing template...")
+	if err := doInstall(ctx, bootstrap.TaskInstall, appName, cmd.Bool("verbose")); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func setupSandboxTemplate(ctx context.Context, cmd *cli.Command) error {
+	// TODO: implement endpoints to fetch sandbox Template for authorized users
+	if true {
+		return errors.New("not implemented")
+	}
+
+	if err := cloneTemplate(ctx, cmd, templateName, appName); err != nil {
+		return err
+	}
+
+	if err := instantiateEnv(ctx, cmd, appName); err != nil {
+		return err
+	}
+
+	if err := doInstall(ctx, bootstrap.TaskInstallSandbox, appName, cmd.Bool("verbose")); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func cloneTemplate(_ context.Context, cmd *cli.Command, templateURL, appName string) error {
+	var cmdErr error
+	if err := spinner.New().
+		Title("Cloning template from " + templateURL).
+		Action(func() {
+			c := exec.Command("git", "clone", "--depth=1", templateURL, appName)
+			var out []byte
+			if out, cmdErr = c.CombinedOutput(); len(out) > 0 && cmd.Bool("verbose") {
+				fmt.Println(string(out))
+			}
+			os.RemoveAll(path.Join(appName, ".git"))
+		}).
+		Run(); err != nil {
+		return err
+	}
+	return cmdErr
+}
+
+func instantiateEnv(ctx context.Context, cmd *cli.Command, rootPath string) error {
+	env := map[string]string{
+		"LIVEKIT_API_KEY":    project.APIKey,
+		"LIVEKIT_API_SECRET": project.APISecret,
+		"LIVEKIT_URL":        project.URL,
+		"LIVEKIT_SANDBOX":    "TODO",
+	}
+
+	prompt := func(key, oldValue string) (string, error) {
+		var newValue string
+		if err := huh.NewInput().
+			Title("Enter " + key + "?").
+			Placeholder(oldValue).
```

Comment:
re: demo, maybe this should be a password field? it's being pasted anyway

```diff
			Placeholder(oldValue).
+ 			EchoMode(textinput.EchoPassword).
```

Reply:
Solid idea. I leaked my keys 5 ways in that demo 😆

Reply:
At the same time, not all of these are passwords/secrets. Our livestream example app actually uses an environment variable that stores a whole yaml document with a UI configuration. That could be quite annoying. Maybe we just trust the devs to have some opsec?

Reply:
hrm, yeah. maybe a heuristic that searches for `SECRET` or `KEY` or `PASSWORD`?